spark_log_parser/parsing_models/application_model_v2.py

```diff
@@ -478,15 +492,27 @@ def compute_executor_info(
         df = defaultdict(lambda: [])
         for xid, executor in app_model.executors.items():
 
-            # Special case for handling end_time
-            if executor.end_time is not None:
-                end_time = executor.end_time / 1000 - app_model.start_time
-            else:
-                end_time = executor.end_time
+            # There appears to be a scenario in Spark where if an executor is started and then stopped very quickly,
+            #  that executor's startup routine will cause an exception, killing the executor. Since the startup routine
+            #  never actually finished, a `start_time` is never registered. Databricks Autoscaling or Spot Terminations
+            #  are common causes of this scenario.
+            start_time = (
+                (executor.start_time / 1000) - app_model.start_time
+                if executor.start_time is not None
+                else None
+            )
+
+            # If an executor lives until the end of the Spark Application, it may never register an `end_time`
+            # TODO - should be app_model.end_time for the default value?
+            end_time = (
+                (executor.end_time / 1000) - app_model.start_time
+                if executor.end_time is not None
+                else None
+            )
 
             df["executor_id"].append(xid)
             df["cores"].append(executor.cores)
-            df["start_time"].append(executor.start_time / 1000 - app_model.start_time)
+            df["start_time"].append(start_time)
```

Comment:
Sorry, lots of autoformatting in this diff... this is the logic change to address the problem